platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/ImplementationSnapshotSerializer.java

```diff
@@ -94,7 +94,7 @@ public ImplementationSnapshot read(Decoder decoder) throws Exception {
             return readAdditionalData(classIdentifier, decoder);
         }
 
-        protected final Serializer<HashCode> hashCodeSerializer = new HashCodeSerializer();
+        protected final HashCodeSerializer hashCodeSerializer = new HashCodeSerializer();
```

Comment:
💭 ErrorProne suggests all Enum fields to be immutable. It makes no sense to mark all `Serializer` implementations as such, so I'm using a concrete type here.